sway/input/cursor.c

```diff
@@ -39,19 +36,42 @@ static void cursor_send_pointer_motion(struct sway_cursor *cursor,
 			};
 
 			if (wlr_box_contains_point(&box, cursor->x, cursor->y)) {
-				surface = xsurface->surface;
-				sx = cursor->x - box.x;
-				sy = cursor->y - box.y;
-				wlr_seat_pointer_notify_enter(seat, surface, sx, sy);
-				wlr_seat_pointer_notify_motion(seat, time, sx, sy);
-				return;
+				*surface = xsurface->surface;
+				*sx = cursor->x - box.x;
+				*sy = cursor->y - box.y;
+				return view->swayc;
 			}
 		}
 	}
 
-	struct sway_container *swayc =
-		container_at(&root_container, cursor->x, cursor->y, &surface, &sx, &sy);
-	if (swayc) {
+	// find the output the cursor is on
+	struct wlr_output_layout *output_layout =
+		root_container.sway_root->output_layout;
+	struct wlr_output *wlr_output =
+		wlr_output_layout_output_at(output_layout, cursor->x, cursor->y);
+	if (wlr_output == NULL) {
+		return NULL;
+	}
+	struct sway_output *output = wlr_output->data;
+
+	// find the focused workspace on the output for this seat
+	struct sway_container *workspace =
+		sway_seat_get_focus_inactive(cursor->seat, output->swayc);
+	if (workspace->type != C_WORKSPACE) {
+		workspace = container_parent(workspace, C_WORKSPACE);
+	}
+
+	return container_at(workspace, cursor->x, cursor->y, surface, sx, sy);
+}
+
+static void cursor_send_pointer_motion(struct sway_cursor *cursor,
+		uint32_t time) {
+	struct wlr_seat *seat = cursor->seat->wlr_seat;
+	struct wlr_surface *surface = NULL;
+	double sx, sy;
+	struct sway_container *cont = cursor_at(cursor, &surface, &sx, &sy);
+
+	if (cont) {
```

Comment:
The way this will work (like it works in rootston), you need to handle the case where you get a container but no surface. In this case, the user has clicked on the decorations of the window. The surface should receive keyboard focus but not get any pointer events.

Reply:
Done